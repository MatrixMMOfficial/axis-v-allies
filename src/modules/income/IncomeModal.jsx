import React from 'react'
import { Link } from 'react-router-dom'

const IncomeModal = ({ currentPower, npl, objectives, nextNpl }) => {
  return (
    <div>
      <h1>Collect Income</h1>
      <p>
        Since your last turn you have maintained control of territories with a total value of {npl} <abbr title="National Production Level">N.P.L.</abbr></p>
      <h3>{currentPower.name} National Objectives</h3>
      <p>Your forces have made progress towards the following national objectives:</p>
      <NationalObjectives objectives={objectives} />
      <p> so you will be able to use {nextNpl} <abbr title="Industrial Production Certificates">I.P.C.s</abbr> next turn</p>
      <nav>
        <Link to="/purchase" className="btn">Back</Link>
        <Link to="/plan-combat" className="btn">Plan Combat</Link>
      </nav>
    </div>
  )
}

export default IncomeModal

const NationalObjectives = ({ objectives }) => {
  return (
    <ul>
      {objectives.map((objective, index) => <Objective 
        key={index}
        text={objective.text} 
        value={objective.value} />)}
    </ul>
  )
}

const Objective = ({ text, value }) => {
  return (
    <li>{text}: <strong>{value} I.P.C.s</strong></li>
  )
}

