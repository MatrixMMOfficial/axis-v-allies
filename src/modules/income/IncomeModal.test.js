'use strict'

import React from 'react'
import { MemoryRouter } from 'react-router'
import IncomeModal from './IncomeModal'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const objectives = [ 
    { text: 'goal 1', value: 1 }, { text: 'goal 2', value: 2 }
  ]
  const tree = renderer.create(
    <MemoryRouter>
      <IncomeModal 
        currentPower={ { name: 'Germany' } }
        npl={60}
        objectives={objectives}
        nextNpl={63}
        setIncome={() => {}}
      />
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

