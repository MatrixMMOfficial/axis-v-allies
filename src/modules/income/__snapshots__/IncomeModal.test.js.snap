// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Collect Income
  </h1>
  <p>
    Since your last turn you have maintained control of territories with a total value of 
    60
     
    <abbr
      title="National Production Level"
    >
      N.P.L.
    </abbr>
  </p>
  <h3>
    Germany
     National Objectives
  </h3>
  <p>
    Your forces have made progress towards the following national objectives:
  </p>
  <ul>
    <li>
      goal 1
      : 
      <strong>
        1
         I.P.C.s
      </strong>
    </li>
    <li>
      goal 2
      : 
      <strong>
        2
         I.P.C.s
      </strong>
    </li>
  </ul>
  <p>
     so you will be able to use 
    63
     
    <abbr
      title="Industrial Production Certificates"
    >
      I.P.C.s
    </abbr>
     next turn
  </p>
  <nav>
    <a
      className="btn"
      href="/purchase"
      onClick={[Function]}
    >
      Back
    </a>
    <a
      className="btn"
      href="/plan-attacks"
      onClick={[Function]}
    >
      Plan Combat
    </a>
  </nav>
</div>
`;
